src/main/java/io/r2dbc/postgresql/codec/Point.java

```diff
@@ -0,0 +1,118 @@
+package io.r2dbc.postgresql.codec;
+
+/**
+ * <p>It maps to the point datatype in org.postgresql.</p>
+ *
+ * <p>This implements a version of java.awt.Point, except it uses double to represent the coordinates.</p>
+ */
+public class Point {
+    private double x;
+    private double y;
+
+    /**
+     * @param x coordinate
+     * @param y coordinate
+     */
+    public Point(double x, double y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    public double getX() {
+        return x;
+    }
+
+    public void setX(double x) {
+        this.x = x;
+    }
+
+    public double getY() {
+        return y;
+    }
+
+    public void setY(double y) {
+        this.y = y;
+    }
+
+    /**
+     * Translate the point by the supplied amount.
+     *
+     * @param x integer amount to add on the x axis
+     * @param y integer amount to add on the y axis
+     */
+    public void translate(int x, int y) {
```

Comment:
Those mutation-like methods should return a new instance of `Point`. Applying mutation to objects always opens up paths for potential bugs.

Reply:
These methods were present in original driver, so I added it as it is. Do you think these methods should be part of this class now or should I remove them?

Reply:
Using PGJDBC as a starting point is fine. With the introduction of geo-types to r2dbc-postgressql, we have a chance to improve the actual design. Let's keep these methods in the type but change these to return a new instance.